### DISCUSSION TITLE
Can not get MapperIgnoreTargetAttribute attributes via reflection

### DISCUSSION THREAD

**Discussion by filiptrivan** [riok/mapperly#1266] [Q&A] [answered]
Whenever I try to get the attributes from the method from the Mapper class, I get only one GeneratedCodeAttribute from which I don't think I can get what I want, is it at all possible to get MapperIgnoreTargetAttribute through reflection and is this expected and correct what is happening to me?

```
        private static MemberInfo[] GetMembersToInclude<ENTITY, MAPPER>()
        {
            Type entityType = typeof(ENTITY);
            Type mapperType = typeof(MAPPER);

            var method = mapperType.GetMethods()
                .Where(m => m.Name == "MapExcel" && m.ReturnType == entityType)
                .Single();

            IEnumerable<MapperIgnoreTargetAttribute> ignoringAttributesFromMapperClass = method.GetCustomAttributes<MapperIgnoreTargetAttribute>();

            MemberInfo[] memberInfos = entityType
                .GetProperties(BindingFlags.Instance | BindingFlags.Public)
                .Where(prop => ignoringAttributesFromMapperClass.Where(att => att.Target == prop.Name).Any() == false)
                .ToArray();

            return memberInfos;
        }
```

**Comment by latonz** [comment, marked as answer]
By default the Mapperly attributes are conditional and are not present at runtime since they are not needed at runtime. However, if you really need them you can preserve them, see the [docs](https://mapperly.riok.app/docs/getting-started/installation/).
Just for my understanding: Can you elaborate on your use-case and why you need access to these attributes at runtime?
